modules/dnn/CMakeLists.txt

```diff
@@ -240,6 +249,10 @@ ocv_create_module(${libs} ${dnn_runtime_libs})
 ocv_add_samples()
 ocv_add_accuracy_tests(${dnn_runtime_libs})
 
+if(NOT BUILD_PROTOBUF)
+  ocv_target_compile_definitions(opencv_test_dnn PRIVATE "OPENCV_DNN_EXTERNAL_PROTOBUF=1")
+endif()
```

Comment:
Does it mean that `DISABLED_Interp` was always disabled? There was `#ifndef OPENCV_DNN_EXTERNAL_PROTOBUF` but no definition?

Reply:
`OPENCV_DNN_EXTERNAL_PROTOBUF` definition seems not be available for `opencv_test_dnn` in currently implementation.

Old test result with external protobuf (not including this change) is following. If that definition is available, test for `Interp` is disabled. But the test  is executed.

So I think an additional fix is needed.

Code

https://github.com/opencv/opencv/blob/590f150d5e032165e27d81294c9b7ac710b77f11/modules/dnn/test/test_layers.cpp#L1654-L1658

Result

```plaintext
[       OK ] Test_Caffe_layers.PriorBox_squares/0 (0 ms)
[ RUN      ] Test_Caffe_layers.Interp/0, where GetParam() = OCV/CPU
E0000 00:00:1696641648.291055   46844 text_format.cc:433] Error parsing text-format opencv_caffe.NetParameter: 14:16: Message type "opencv_caffe.LayerParameter" has no field named "interp_param".
unknown file: Failure
C++ exception with description "OpenCV(4.8.0-dev) /home/kmtr/work/opencv/modules/dnn/src/caffe/caffe_io.cpp:1162: error: (-2:Unspecified error) FAILED: ReadProtoFromTextFile(param_file, param). Failed to parse NetParameter file: /home/kmtr/work/opencv_extra/testdata/dnn/layers/layer_interp.prototxt in function 'ReadNetParamsFromTextFileOrDie'
" thrown in the test body.
[  FAILED  ] Test_Caffe_layers.Interp/0, where GetParam() = OCV/CPU (1 ms)
[----------] 31 tests from Test_Caffe_layers (93 ms total)

[----------] Global test environment tear-down
[==========] 31 tests from 1 test case ran. (93 ms total)
[  PASSED  ] 26 tests.
```

Reply:
Good catch